tests/test_patroni.py

```diff
@@ -236,3 +241,257 @@ def test_ensure_unique_name(self):
             # Only if the api of the running node is reachable do we throw an error
             with patch.object(self.p, 'request', Mock()):
                 self.assertRaises(SystemExit, self.p.ensure_unique_name)
+
+
+@patch('patroni.psycopg.connect', psycopg_connect)
+@patch('socket.getaddrinfo', Mock(return_value=[(0, 0, 0, 0, ('1.9.8.4', 1984))]))
+@patch('builtins.open', MagicMock())
+@patch('socket.gethostname', Mock(return_value='testhost'))
+@patch('subprocess.check_output', Mock(return_value=b"postgres (PostgreSQL) 13.2"))
+@patch('psutil.Process.exe', Mock(return_value='/bin/dir/from/running/postgres'))
+@patch('psutil.Process.__init__', Mock(return_value=None))
+class TestGenerateConfig(unittest.TestCase):
```

Comment:
Should the tests for `config_generator.py` be in its own `test_config_generator.py` file?